### PR TITLE
move to pyproject.toml

### PR DESCRIPTION
**Pull request by martinvonk**
I'm getting issues installing Metran using the old setup.py method (e.g. #18). So I've created a pyproject.toml just like Pastas.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -14,11 +14,12 @@ formats: all
 
 # Optionally set the version of Python and requirements required to build your docs
 python:
-  version: 3.7
+  version: "3.7"
   install:
-    - requirements: docs/requirements.txt
     - method: pip
       path: .
+      extra_requirements:
```
Comment (dbrakenhoff) [inline .readthedocs.yml:21]: rmeove

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "Programming Language :: Python :: 3.9",
+    "Topic :: Scientific/Engineering :: Hydrology",
+]
+
+[project.urls]
+homepage = "https://github.com/pastas/metran"
+repository = "https://github.com/pastas/metran"
+documentation = "https://metran.readthedocs.io/en/latest/"
+
+[project.optional-dependencies]
```
[the remark this answers is not in the document]
Comment (dbrakenhoff) [reply pyproject.toml:43]: add ci section

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -28,7 +28,6 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install flake8 pytest
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
           pip install codecov
```
Comment (dbrakenhoff) [inline .github/workflows/ci.yml:31]: move these to ci under [project.optional_dependencies]

**Review by dbrakenhoff — CHANGES_REQUESTED** [review]
see comments

File: pyproject.toml [hunk 4]
```diff
@@ -41,8 +41,10 @@ repository = "https://github.com/pastas/metran"
 documentation = "https://metran.readthedocs.io/en/latest/"
 
 [project.optional-dependencies]
-rtd = ["metran"]
-dev = ["metran[rtd]"]
+linting = ["flake8"]
+pytesting = ["pytest"]
+coveraging = ["codecov", "pytest-cov", "pytest-dependency", "codacy-coverage"]
+ci = ["metran[linting,pytsting,coveraging]"]
```
[the remark this answers is not in the document]
Comment (dbrakenhoff) [reply pyproject.toml:47]: pytest-dependency should formally be part of the pytesting category

[the remark this answers is not in the document]
Comment (martinvonk) [reply pyproject.toml:47]: whoopsi

**Pull request merged**